torch/distributed/_tensor/ops/_matrix_ops.py

```diff
@@ -161,7 +162,9 @@ def baddmm_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
     return _addmm_like_strategy("bmk,bkn->bmn", mesh, op_schema)
 
 
-@register_op_strategy(aten._scaled_dot_product_flash_attention.default)
+@register_op_strategy(
+    aten._scaled_dot_product_flash_attention.default, schema_info=RuntimeSchemaInfo(5)
```

Comment:
In addition to the arg `bool return_debug_mask` at index position 5, do you think we should also avoid cache hit on the arg `bool is_causal` at index position 4?

Reply:
Since we are only concerned with sharding propagation op caching and the sharding logic does not appear conditioned in any way on `is_causal`, I think it would be preferable to start with arg idx 5 and only move to 4 if the propagation logic is altered to be conditioned on `is_causal` (as it currently is with [`return_debug_mask`](https://github.com/pytorch/pytorch/blob/629bd6f7184da58e21ba1580efbc8ff0c24dad14/torch/distributed/_tensor/ops/_matrix_ops.py#L203-L207) for instance).

Reply:
makes sense to me